xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -830,69 +848,116 @@ int CMusicInfoScanner::RetrieveMusicInfo(const std::string& strDirectory, CFileI
         m_musicDatabase.SetArtForItem(artist.idArtist, MediaTypeArtist, GetArtistArtwork(artist));
       }
     }
+    numAdded += album->songs.size();
+  }
+
+  if (m_handle)
+    m_handle->SetTitle(g_localizeStrings.Get(505));
+
+  return numAdded;
+}
+
+void MUSIC_INFO::CMusicInfoScanner::ScrapeInfoAddedAlbums()
+{
+  /* Strategy: Having scanned tags, make a list of albums and add them to the library, only then try
+  to scrape additional album and artist information. Music is often tagged to a mixed standard
+  - some albums have mbid tags, some don't. Once all the musc files have been added to the library,
```

Comment:
+  - some albums have mbid tags, some don't. Once all the music files have been added to the library,